page/renderer/ComponentUtils.js

```diff
@@ -0,0 +1,100 @@
+export default class ComponentUtils {
+  /**
+   * Initializes the registry used for managing component utils.
+   *
+   * @param {ObjectContainer} oc The application's dependency injector - the
+   *        object container.
+   */
+  constructor(oc) {
+    /**
+     * The application's dependency injector - the object container.
+     *
+     * @type {ObjectContainer}
+     */
+    this._oc = oc;
+
+    /**
+     * Map of registered utilities.
+     *
+     * @type {Object<string, function(new: T, ...*)|function(...*): T>}
+     */
+    this._utilityClasses = {};
+
+    /**
+     * Map of instantiated utilities
+     *
+     * @type {Object<string, Object>}
+     */
+    this._utilities = null;
+  }
+
+  /**
+   * Registers single utility class or multiple classes in alias->class mapping.
+   *
+   * @param {string|Object<string, function(new: T, ...*)|function(...*): T>} name
+   * @param {function(new: T, ...*)|function(...*): T} componentUtilityClass
+   */
+  register(name, componentUtilityClass) {
+    if (typeof componentUtilityClass === 'function') {
+      const alias = String(name);
+      this._utilityClasses[alias] = componentUtilityClass;
+
+      if (this._utilities) {
+        this._createUtilityInstance(alias, componentUtilityClass);
+      }
+    } else if (
+      name &&
+      typeof name === 'object' &&
+      name.constructor === Object
+    ) {
+      const utilityClasses = name;
+
+      for (const alias of Object.keys(utilityClasses)) {
+        if (!utilityClasses.hasOwnProperty(alias)) {
+          continue;
+        }
+
+        this._utilityClasses[alias] = utilityClasses[alias];
```

Comment:
why don't you call method `register` with `alias` and `utilityClasses[alias]`?

Reply:
Didn't thought about it. Fixed.